src/test/integration/services/user.spec.js

```diff
@@ -0,0 +1,199 @@
+  })
+
+  test('should throw error for update on non-existing user', async () => {
+    const nonExistingId = mongoose.Types.ObjectId()
+
+    await expect(userService.updateUser(nonExistingId, 'student', { firstName: 'updatedName' })).rejects.toThrowError(
+      'User with the specified ID was not found.'
+    )
+  })
+
+  test('should update user status', async () => {
+    const createdUser = await createUser(userData)
+
+    const updateStatus = { student: 'inactive' }
+
+    await userService.updateStatus(createdUser._id, updateStatus)
+
+    const updatedUser = await userService.getUserById(createdUser._id)
+    expect(updatedUser.status.student).toBe('inactive')
+  })
+
+  test('should throw error for status update on non-existing user', async () => {
+    const nonExistingId = mongoose.Types.ObjectId()
+
+    await expect(userService.updateStatus(nonExistingId, { student: 'inactive' })).rejects.toThrowError(
+      'User with the specified ID was not found.'
+    )
+  })
+
+  test('should delete user', async () => {
+    const createdUser = await createUser(userData)
+
+    await userService.deleteUser(createdUser._id)
+
+    const deletedUser = await userService.getUserById(createdUser._id)
+    expect(deletedUser).toBeNull()
+  })
+  test('should throw error for duplicate email during user creation', async () => {
+    await createUser(userData)
+
+    await expect(createUser(userData)).rejects.toThrowError('User with the specified email already exists.')
+  })
+
+  test('should update user with allowed fields only', async () => {
+    const createdUser = await createUser(userData)
+
+    const updateData = {
+      firstName: 'UpdatedName',
+      lastName: 'UpdatedLastName',
+      invalidField: 'Should not be allowed'
+    }
+
+    await userService.updateUser(createdUser._id, 'student', updateData)
+
+    const updatedUser = await userService.getUserById(createdUser._id)
+    expect(updatedUser.firstName).toBe('UpdatedName')
+    expect(updatedUser.lastName).toBe('UpdatedLastName')
+    expect(updatedUser.invalidField).toBeUndefined()
+  })
+
+  test('should delete user and return null when fetching deleted user', async () => {
+    const createdUser = await createUser(userData)
+
+    await userService.deleteUser(createdUser._id)
+
+    const deletedUser = await userService.getUserById(createdUser._id)
+    expect(deletedUser).toBeNull()
+  })
+
+  test('should handle updating user status for different roles', async () => {
+    const createdUser = await createUser(userData)
+
+    const updateStatus = { student: 'inactive', tutor: 'active' }
+
+    await userService.updateStatus(createdUser._id, updateStatus)
+
+    const updatedUser = await userService.getUserById(createdUser._id)
+    expect(updatedUser.status.student).toBe('inactive')
+    expect(updatedUser.status.tutor).toBe('active')
+  })
+
+  test('createError should create error with specific status and message', () => {
```

Comment:
it looks like this test should be placed somewhere else, because it does not test the functionality of  User